envtrace-helper.c

```diff
@@ -1,4 +1,11 @@
+#if defined(__linux__) && !defined(_GNU_SOURCE)
 #define _GNU_SOURCE
+#endif
+
+#if !defined(RTLD_NEXT)
+#define RTLD_NEXT 0
```

Comment:
This looks fragile to me.

`man dlsym()` says:

```
The symbols RTLD_DEFAULT and RTLD_NEXT are defined by <dlfcn.h> only when _GNU_SOURCE was defined before including it.
```

Can we just replace these few lines with an unconditional `#define _GNU_SOURCE 1`?